File: Frontend/src/app/models/report.ts

export class Report
{
  public reportId: number;

  public stationId: number;

  public stationName: string;

  public configId: number;

  public configName: string;

  public reportTime: string;

  public backupSize: number;

  public success: boolean;

  public constructor(reportId: number, stationId: number, stationName: string, configId: number, configName: string, reportTime: string, backupSize: number, success: boolean) {
    this.reportId = reportId;
    this.stationId = stationId;
    this.configId = configId;
    this.reportTime = reportTime;
    this.backupSize = backupSize;
    this.success = success;
  }

}

File: Backend/Models/Report.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BackupSystem.Models;

public partial class Report
{
    public int ReportId { get; set; }
    public int StationId { get; set; }
    public int ConfigId { get; set; }
    public DateTime ReportTime { get; set; }
    public ulong BackupSize { get; set; }
    public Boolean Success { get; set; }
    public string ErrorMessage { get; set; } = string.Empty;

    [ForeignKey("ConfigId")]
    public virtual Configuration Config { get; set; } = null!;

    public virtual Station Station { get; set; } = null!;
}
